mbed/mbed.py

```diff
@@ -3011,6 +3011,51 @@ def help_():
     return parser.print_help()
 
 
+@subcommand('unittest',
+    dict(name='--compile', action='store_true', dest="compile_only", help='Only compile unit tests'),
+    dict(name='--run', action='store_true', dest="run_only", help='Only run unit tests'),
+    dict(name=['-c', '--clean'], action='store_true', help='Clean the build directory'),
+    dict(name=['-d', '--debug'], action='store_true', help='Enable debug build'),
+    dict(name='--coverage', choices=['html', 'xml', 'both'], help='Generate code coverage report'),
+    dict(name=['-m', '--make-program'], choices=['gmake', 'make', 'mingw32-make', 'ninja'], help='Which make program to use'),
+    dict(name=['-g', '--generator'], choices=['Unix Makefiles', 'MinGW Makefiles', 'Ninja'], help='Which CMake generator to use'),
+    dict(name=['-r', '--regex'], help='Run tests matching regular expression'),
+    dict(name='--build', help='Build directory. Default: mbed-os/UNITTESTS/build/'),
```

Comment:
This default build directory is outside of the normal root `BUILD` directory, is this something that has already been agreed upon? @theotherjimmy

Reply:
Since unit tests use completely different build environment, they will not be part of mbed-os application and the tools used are separate from other mbed-os tools it would be better to keep the build folders separate in my opinion.

Reply:
@lorjala I'm concerned that your build directory will be scanned by the build tools. How are you avoiding that?

Reply:
@theotherjimmy UNITTESTS directory has .mbedignore file, but it could be a possibility to add .mbedignore into the build directory as well during the folder creation if that is the concern.

Reply:
😉 Defaults to `BUILD/unittests` for now. Let's see how it works.